core/src/main/java/org/jdbi/v3/core/internal/JdbiClassUtils.java

```diff
@@ -192,6 +195,8 @@ public static <T> T findConstructorAndCreateInstance(Class<T> type, Class<?>[] t
         return findConstructorAndCreateInstance(type, types, DEFAULT_EXCEPTION_HANDLER, parameters);
     }
 
+    private static final Map<Class<?>, CheckedFunction<Object[], Object>> CTOR_CACHE = new ConcurrentHashMap<>();
```

Comment:
This will cause Jdbi to retain a reference to any class it observes forever, preventing class unloading.
Probably most users don't care but it could be considered a memory leak.